app/coffee/HttpController.coffee

```diff
@@ -103,20 +103,28 @@ module.exports = HttpController =
 			logger.log project_id: project_id, doc_id: doc_id, "flushed doc via http"
 			res.send 204 # No Content
 
-	flushAndDeleteDoc: (req, res, next = (error) ->) ->
+	deleteDoc: (req, res, next = (error) ->) ->
 		doc_id = req.params.doc_id
 		project_id = req.params.project_id
-		logger.log project_id: project_id, doc_id: doc_id, "deleting doc via http"
+		flush = req.body.flush ? true
+		logger.log project_id: project_id, doc_id: doc_id, flush: flush, "deleting doc via http"
 		timer = new Metrics.Timer("http.deleteDoc")
-		DocumentManager.flushAndDeleteDocWithLock project_id, doc_id, (error) ->
-			timer.done()
-			# There is no harm in flushing project history if the previous call
-			# failed and sometimes it is required
-			HistoryManager.flushProjectChangesAsync project_id
+		if flush
+			DocumentManager.flushAndDeleteDocWithLock project_id, doc_id, (error) ->
+				timer.done()
+				# There is no harm in flushing project history if the previous call
+				# failed and sometimes it is required
+				HistoryManager.flushProjectChangesAsync project_id
 
-			return next(error) if error?
-			logger.log project_id: project_id, doc_id: doc_id, "deleted doc via http"
-			res.send 204 # No Content
+				return next(error) if error?
+				logger.log project_id: project_id, doc_id: doc_id, "deleted doc via http"
+				res.send 204 # No Content
+		else
+			DocumentManager.deleteDocWithLock project_id, doc_id, (error) ->
```

Comment:
I think it is worth a comment warning that this action is destructive and the cases where you might need to use it.